var/spack/repos/builtin/packages/py-pennylane-lightning/package.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack.package import *
+
+
+class PyPennylaneLightning(PythonPackage):
+    """PennyLane-Lightning plugin"""
+
+    homepage = "https://github.com/PennyLaneAI/pennylane-lightning"
+    url = "https://github.com/PennyLaneAI/pennylane-lightning/archive/refs/tags/v0.28.0.tar.gz"
+    # using github for now, because pypi tarball is missing the CMakeLists.txt file
+    # pypi = "PennyLane-Lightning/PennyLane-Lightning-0.28.0.tar.gz"
+
+    version("0.28.0", sha256="f5849c2affb5fb57aca20feb40ca829d171b07db2304fde0a37c2332c5b09e18")
+
+    depends_on("python@3.8:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+    depends_on("cmake", type="build")
+
+    depends_on("py-ninja", type=("build", "run"))
+    depends_on("py-numpy", type=("build", "run"))
+    # depends_on("py-pennylane@0.19:", type=("build", "run"))  # circular dependency
```

Comment:
This is annoying. I don't know why developers do this, or how pip handles it. @pradyunsg how does pip resolve circular dependencies like this? In Spack we usually just comment out the dep in `setup.py` so that nothing complains.

Reply:
Pinging @josh146 in case he wants to bring in someone from his team, to discuss the requirements for this circular dependency

Reply:
Hey folks, thanks for the work on this. I'll make a follow-up PR here, and tag myself as the maintainer for Lightning. There are a few things I intend to bring in from https://github.com/mlxd/spack/tree/pennylane_lightning, especially around how the package for Lightning is built.